packages/react-components/react-avatar/src/components/AvatarGroupOverflow/AvatarGroupOverflow.test.tsx

```diff
@@ -1,18 +1,61 @@
 import * as React from 'react';
-import { render } from '@testing-library/react';
+import { AvatarGroupItem } from '../AvatarGroupItem/AvatarGroupItem';
 import { AvatarGroupOverflow } from './AvatarGroupOverflow';
+import { avatarGroupOverflowClassNames } from './useAvatarGroupOverflowStyles';
 import { isConformant } from '../../common/isConformant';
+import { RenderResult } from '@testing-library/react';
+
+// testing-library's queryByRole function doesn't look inside portals
+function queryByRoleDialog(result: RenderResult) {
+  const dialogs = result.baseElement.querySelectorAll('*[role="dialog"]');
+  if (!dialogs?.length) {
+    return null;
+  } else {
+    expect(dialogs.length).toBe(1);
+    return dialogs.item(0) as HTMLElement;
+  }
+}
+
+const getOverflowSurfaceElement = (result: RenderResult) => {
+  // Overflow button needs to be clicked otherwise overflowContent won't be rendered.
+  result.queryByRole('button')?.click();
+  return queryByRoleDialog(result)!;
```

Comment:
Don't ignore null here with `!`; check and fail the test if it's null rather than continuing and crashing later.

```suggestion
  const dialog = queryByRoleDialog(result);
  expect(dialog).not.toBeNull();
  return dialog!;
```